pandas/tools/merge.py

```diff
@@ -193,6 +195,13 @@ def __init__(self, left, right, how='inner', on=None,
                 'can not merge DataFrame with instance of '
                 'type {0}'.format(type(right)))
 
+        # warn user when merging between different levels
+        if left.columns.nlevels != right.columns.nlevels:
```

Comment:
Oh, I see. I've been away for some time, but I just went back. I can do a PR for that.